scripts/lib.sh

```diff
@@ -316,3 +379,142 @@ detect_go_version()
 	[ "$?" == "0" ] && [ "$GO_VERSION" != "null" ]
 }
 
+detect_rust_version()
+{
+	info "Detecting agent rust version"
+	typeset -r yq=$(command -v yq || command -v ${GOPATH}/bin/yq)
+	if [ -z "$yq" ]; then
+		source "$yq_file"
+	fi
+
+	local runtimeRevision=""
+
+	# Detect runtime revision by fetching the agent's VERSION file
+	local runtime_version_url="https://raw.githubusercontent.com/kata-containers/agent/${AGENT_VERSION:-master}/VERSION"
+	info "Detecting runtime version using ${runtime_version_url}"
+
+	if runtimeRevision="$(curl -fsSL ${runtime_version_url})"; then
+		[ -n "${runtimeRevision}" ] || die "failed to get agent version"
+		typeset -r runtimeVersionsURL="https://raw.githubusercontent.com/kata-containers/runtime/${runtimeRevision}/versions.yaml"
+		info "Getting rust version from ${runtimeVersionsURL}"
+		# This may fail if we are a kata bump.
+		if RUST_VERSION="$(curl -fsSL "$runtimeVersionsURL" | tac | tac | $yq r - "languages.rust.version")"; then
+			[ "$RUST_VERSION" != "null" ]
+			return 0
+		fi
+	fi
+
+	info "Agent version has not match with a runtime version, assumming it is a PR"
+	local kata_runtime_pkg_dir="${GOPATH}/src/${GO_RUNTIME_PKG}"
+	if [ ! -d "${kata_runtime_pkg_dir}" ];then
+		info "There is not runtime repository in filesystem (${kata_runtime_pkg_dir})"
+		local runtime_versions_url="https://raw.githubusercontent.com/kata-containers/runtime/${KATA_BRANCH}/versions.yaml"
+		info "Get versions file from ${runtime_versions_url}"
+		RUST_VERSION="$(curl -fsSL "${runtime_versions_url}" | tac | tac | $yq r - "languages.rust.version")"
+		if [ "$?" == "0" ] && [ "$RUST_VERSION" != "null" ]; then
+			return 0
+		fi
+
+		return 1
```

Comment:
this looks odd - no error info, and no default return at the end of the function (it's implicit I guess) - but - it seems to be the same as the go version it is copied from, so.... dunno if we want to improve them both. Or, at least a comment on the function noting what it returns, and what that means :-)